assets/js/ViewerApp.js

```diff
@@ -19,17 +21,17 @@ class ViewerApp extends Component {
       .join()
       .receive("ok", response => {console.log("Joined successfully", response)});
 
-    channel.on("new_msg", payload => {
+    channel.on("sign_update", ({sign_id, line_number, text}) => {
       this.setState({
```

Comment:
If you're using setState with this.state, due to the batching of setState, you want to pass a function to setState, not an object.

https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous